cmd/bucket-handlers.go

```diff
@@ -390,11 +390,13 @@ func (api objectAPIHandlers) PutBucketHandler(w http.ResponseWriter, r *http.Req
 		return
 	}
 
-	// Validate if location sent by the client is valid, reject
-	// requests which do not follow valid region requirements.
-	if !isValidLocation(location) {
-		writeErrorResponse(w, ErrInvalidRegion, r.URL)
-		return
+	if !api.gateway {
+		// Validate if location sent by the client is valid, reject
+		// requests which do not follow valid region requirements.
+		if !isValidLocation(location) {
+			writeErrorResponse(w, ErrInvalidRegion, r.URL)
+			return
+		}
```

Comment:
isValidLocation() should be done regardless of gateway mode or not

Reply:
as discussed, this check is needed - hence not removed.